universe/activities/processor.go

```diff
@@ -43,25 +46,30 @@ func (a *Activities) handleNewActivity(activity universe.Activity) error {
 }
 
 func (a *Activities) handleChangedRemovedActivity(activity universe.Activity, updateType posbus.ActivityUpdateType, objectIDs []umid.UMID) error {
-	errCh := make(chan error)
+	var wg sync.WaitGroup
+	errCh := make(chan error, len(objectIDs))
```

Comment:
not a blocker, but since you started using WaitGroup, might as well use `errgroup.Group` instead and avoid the error channel